test/template.js

```diff
@@ -10,63 +10,107 @@ describe('TemplateGenerator', function () {
       templateGenerator = new TemplateGenerator();
     });
 
-    describe('with github link', function () {
+    describe('with named template', function () {
 
-      it('return correct zip filename for https link', function () {
-        let result = templateGenerator.getExternalProject("https://github.com/embark-framework/embark");
-        assert.strictEqual(result.url, "https://github.com/embark-framework/embark/archive/master.zip");
-        assert.strictEqual(result.filePath.replace(/\\/g,'/'), ".embark/templates/embark-framework/embark/master/archive.zip");
-      });
+      it('returns correct info for named template', function () {
+        let result = templateGenerator.getExternalProject("typescript");
```

Comment:
Took me a while to realize this. So you can call `getExternalProject()` with some arbitrary string and then it's gonna try to match that to an `embark-{STRING}-template` repo. 

Just out of curiosity, I assume this is so that we can simply do `--template {SOME_STRING}` and it'll still ideally find a template repo, right?

Maybe it make sense to, at some point, introduce `templateGenerator.getTemplateProject(template: string)`, which will forward to `getExternalProject()` but takes take of expanding something like `typescript` to `embark-typescript-template` beforehand for clarity.

Reply:
Thanks for the insights!